### PR TITLE
PE-7263: Stake-redelegation

### PR DESCRIPTION
**Pull request by kunstmusik**
* Introduces ellipsis button and popup menu to initiate staking, withdrawals, redelegation, expedited withdrawals, and cancel withdrawals. 
* Separates Staking and Withdrawals into separate modals 
* Introduces Redelegation for stake and pending withdrawals. Works between delegate stake and operator stake.
* Removes quick stake button
* Fixes missing Stake and Leave buttons on gateway details page

### CODE REVIEW

File: src/components/modals/OperatorStakingModal.tsx [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
 import Button, { ButtonType } from '../Button';
 import LabelValueRow from '../LabelValueRow';
 import Tooltip from '../Tooltip';
 import ErrorMessageIcon from '../forms/ErrorMessageIcon';
-import {
-  validateIOAmount as validateARIOAmount,
-  validateOperatorWithdrawAmount,
-} from '../forms/validation';
-import { CircleCheckIcon, CircleIcon, InfoIcon } from '../icons';
+import { validateIOAmount as validateARIOAmount } from '../forms/validation';
```
Comment (atticusofsparta) [inline src/components/modals/OperatorStakingModal.tsx:15]: why not rename the actual function?

Comment (kunstmusik) [reply src/components/modals/OperatorStakingModal.tsx:15]: was part of refactoring and didn't catch that auto did that, will update now

File: src/components/modals/RedelegateModal.tsx [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    ? new mARIOToken(targetGateway?.settings.minDelegatedStake)
+        .toARIO()
+        .valueOf()
+    : 10;
+  const minRequiredStakeToAdd =
+    (targetGatewayCurrentStake ?? 0) > 0 ? 1 : minDelegatedStake;
+
+  const validators = useMemo(
+    () => ({
+      redelegationAmount: validateIOAmount(
```
Comment (atticusofsparta) [inline src/components/modals/RedelegateModal.tsx]: Can do a find and replace for validateIOAmount

File: src/components/modals/RedelegateModal.tsx [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+              onChange={(e) => {
+                const textValue = e.target.value;
+
+                if (textValue && isNaN(+e.target.value)) {
+                  return;
+                }
+
+                setAmountToRedelegate(textValue);
+              }}
+            ></input>
```
Comment (atticusofsparta) [inline src/components/modals/RedelegateModal.tsx:261]: nit: <input />

File: src/components/modals/StakeWithdrawalModal.tsx [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+              onChange={(e) => {
+                const textValue = e.target.value;
+
+                if (textValue && isNaN(+e.target.value)) {
+                  return;
+                }
+
+                setAmountToWithdraw(textValue);
+              }}
+            ></input>
```
Comment (atticusofsparta) [inline src/components/modals/StakeWithdrawalModal.tsx:160]: nit: `<input />`

File: src/components/modals/OperatorStakingModal.tsx [hunk 5]
```diff
@@ -1,27 +1,21 @@
...
 import Button, { ButtonType } from '../Button';
 import LabelValueRow from '../LabelValueRow';
 import Tooltip from '../Tooltip';
 import ErrorMessageIcon from '../forms/ErrorMessageIcon';
-import {
-  validateIOAmount as validateARIOAmount,
-  validateOperatorWithdrawAmount,
-} from '../forms/validation';
-import { CircleCheckIcon, CircleIcon, InfoIcon } from '../icons';
+import { validateARIOAmount as validateARIOAmount } from '../forms/validation';
```
Comment (atticusofsparta) [inline src/components/modals/OperatorStakingModal.tsx:15]: no need to do `as`

**Review by atticusofsparta — APPROVED** [review]

File: src/components/modals/RedelegateModal.tsx (hunk 2 above)
Comment (kunstmusik) [reply src/components/modals/RedelegateModal.tsx]: done

File: src/components/modals/RedelegateModal.tsx (hunk 3 above)
Comment (kunstmusik) [reply src/components/modals/RedelegateModal.tsx]: done

File: src/components/modals/StakeWithdrawalModal.tsx (hunk 4 above)
Comment (kunstmusik) [reply src/components/modals/StakeWithdrawalModal.tsx]: done

**Pull request merged**